src/columns/Enum.jl

```diff
@@ -0,0 +1,73 @@
+is_ch_typename(::Val{:Enum8})  = true
+is_ch_typename(::Val{:Enum16})  = true
+
+const ENUM_RE_ARG = r"""
+
+           '((?:(?:[^'])|(?:\\'))+)'
+           \s*=\s*
+           (-?\d+)
+           \s*$
+       """x
+
+function make_enum_map(::Type{BaseT}, args...) where {BaseT}
+    map = Dict{String, BaseT}()
+    for arg in args
+        m = match(ENUM_RE_ARG, arg)
+        isnothing(m) && error("Wrong enum argument $arg")
+        map[m.captures[1]] = parse(BaseT, m.captures[2])
+    end
+    return map
+end
+
+function read_enum_data(sock::ClickHouseSock, num_rows::VarUInt, ::Type{BaseT}, args...) where {BaseT}
+    map = make_enum_map(BaseT, args...)
+    levels = collect(keys(map))
+    enum_to_level = Dict(zip(values(map), 1:length(map)))
+    data = chread(sock, Vector{BaseT}, num_rows)
+    result = CategoricalVector{String, BaseT}(undef, 0, levels = levels)
```

Comment:
> Enum is now read as CategorialVector. Anyone can use both CategorialVector and Vector{String} when writing.

This actually was already the case previously; old code:
```julia
inv_map = Dict(v => k for (k, v) ∈ enum_def)
data = recode(data, inv_map...)
```

`recode` is a `CategoricalArrays` function that essentially does exactly what you are doing manually here. It's a bit hard to find because the name doesn't really convey what it does very well. I remember doing exactly what you did here until I accidentally stumbled over `recode`. :) 

See:
https://juliadata.github.io/CategoricalArrays.jl/v0.5/using.html#CategoricalArrays.recode

Reply:
This is not true:
```julia
julia> using CategoricalArrays

julia> a = [1,3,1,3,2]
5-element Array{Int64,1}:
 1
 3
 1
 3
 2

julia> b = recode(a, 1=>"a", 2=>"b", 3=>"c")
5-element Array{String,1}:
 "a"
 "c"
 "a"
 "c"
 "b"

julia> typeof(b)
Array{String,1}
```
if you look to the source code here https://github.com/JuliaData/CategoricalArrays.jl/blob/579e1e1fbea517815ea90f489b98d79d1ff6d484/src/recode.jl#L269-L331 you will see that the `recode(::AbstractVector{}, ...)` returns a Vector

Reply:
Oh lol, yeah true. So I guess you unintentionally fixed a bug that I didn't know exists then — I had previously assumed that this was already a categorical array and because semantics of cat arrays and regular arrays are so similar, I never realized this.

Doing
```julia-repl
julia> recode(CategoricalArray([1, 2, 3]), 1 => 'a', 2 => 'b', 3 => 'c')
3-element CategoricalArray{Char,1,UInt32}:
 'a'
 'b'
 'c'
```
makes it return a `CategoricalArray`, but your code is probably more efficient because it doesn't first copy things into a `CategoricalArray`.